panels/metrics/cc-metrics-panel.c

```diff
@@ -190,25 +179,29 @@ cc_metrics_panel_constructed (GObject *object)
                         G_CALLBACK (on_metrics_proxy_properties_changed), self);
 
       value = g_dbus_proxy_get_cached_property (self->metrics_proxy, "Enabled");
-      self->metrics_active = g_variant_get_boolean (value);
+      metrics_active = g_variant_get_boolean (value);
       g_clear_pointer (&value, g_variant_unref);
     }
 
   permission = polkit_permission_new_sync ("com.endlessm.Metrics.SetEnabled",
                                            NULL, NULL, NULL);
-  if (!permission)
-    metrics_can_change = FALSE;
-  else
+  if (permission)
     metrics_can_change = g_permission_get_allowed (permission);
 
   box = gtk_box_new (GTK_ORIENTATION_HORIZONTAL, 6);
-  gtk_widget_show (box);
+  gtk_widget_set_visible (box, true);
 
   gtk_widget_set_sensitive (GTK_WIDGET (self->enable_metrics_switch), metrics_can_change);
-  gtk_switch_set_state (GTK_SWITCH (self->enable_metrics_switch), self->metrics_active);
+
+  /* It would be clearer to use our own metrics_enabled property, but for this
```

Comment:
Is there such a property?

Reply:
There isn't. I rearranged this in such a way that it's more clear what is going on, so that awkward comment is unnecessary :)